solr/contrib/blob-directory/src/java/org/apache/solr/blob/BlobDirectory.java

```diff
@@ -181,7 +182,12 @@ protected void ensureOpen() throws AlreadyClosedException {
     @Override
     public void close() throws IOException {
       // Free the reference to the IndexOutput.
-      blobFileSupplier.indexOutput = null;
+      blobFileSupplier.getBlobFile();
```

Comment:
Calling getBlobFile frees a reference?

Reply:
I agree it's not clear. I'll separate freeing the IndexOutput reference and getting the BlobFile (the latter is done with the IndexOutput to get the file length and checksum).